lib/translator.js

```diff
@@ -1,80 +1,167 @@
 var YAML = require('yamljs');
 var q = require('q');
+var fs = require('fs');
 
-module.exports = (function() {
+var DEFAULT_REGEXP = /\{{2}([\w\.\s\"\']+\s?\|\s?translate[\w\s\|]*)\}{2}/g;
 
-  var EXPRESSION_REG = /\{{2}([\w\.\s\"\']+)\s?\|\stranslate([\w\s\|]*)\}{2}/g;
-  var FILTER_REG = /translate(([\|\s]*)([\w]+))*/;
-  var EXPRESSION_VALUE_REG = /\{{2}\s*[\'\"]?([\w\.]*)[\'\"]?/;
 
-  var getLocale = function(dir){
-    return YAML.load(dir);
-  };
+var baseTransforms = {
+  translate: function(content, dictionary){
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    return content.trim().split('.').reduce(function(dict, key){
+        if (!dict) {
+          return null;
+        }
+        return dict[key];
+      }, dictionary) ||
+      new Error('No such content (' + content + ') at dictionary.');
+  },
+
+  uppercase: function(content) {
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    return content.toUpperCase();
+  },
 
-  var getLocaleValue = function(key, locale) {
-    return key.trim().split('.').reduce(function(obj, key){
-      return obj[key];
-    }, locale);
-  };
+  lowercase: function(content) {
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    return content.toLowerCase();
+  },
 
-  // FILTERS
+  capitalize: function(content){
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    return content.charAt(0).toUpperCase() + content.substring(1).toLowerCase();
+  },
 
-  var uppercase = function(content) {
-    return content.toUpperCase();
-  };
+  capitalizeEvery: function(content){
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    return content.toLowerCase().replace(/[^\s\.!\?]+(.)/g, function(word){
+      return this.capitalize(word);
+    });
+  },
+
+  reverse: function(content){
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    var res = '';
```

Comment:
What do you think of using reduce here in a following way:

```javascript
return content.split('').reduce(function(r, c) { return c+r;}, '')
```
? 

Sorry for nitpicking